cmds/core/timeout/main.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2012-2022 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Run a command and kill it if it runs more than a specified duration
+//
+// Synopsis:
+//	timeout [-t duration-string] command [args...]
+//
+// Description:
+//	timeout will run the command until it succeeds or too much time has passed.
+//	The default timeout is 30s.
+//	If no args are given, it will print a usage error.
+//
+// Example:
+//	$ timeout echo hi
+//	hi
+//	$
+//	$./timeout -t 5s bash -c 'sleep 40'
+//	$ 2022/03/31 14:47:32 signal: killed
+//	$./timeout  -t 5s bash -c 'sleep 40'
+//	$ 2022/03/31 14:47:40 signal: killed
+//	$./timeout  -t 5s bash -c 'sleep 1'
+//	$
+
+//go:build !test
+// +build !test
+
+package main
+
+import (
+	"context"
+	"errors"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"time"
+)
+
+type cmd struct {
+	args         []string
+	timeout      string
+	in, out, err *os.File
+}
+
+var (
+	timeout        = flag.String("t", "30s", "Timeout for command")
```

Comment:
if you use `flag.Duration("t", 30 * time.Second, "Timeout for command")`  then you can get rid of the `time.ParseDuration` call (and change cmd.timeout to be of time `time.Duration`)

Reply:
I love it. I did not even know about flag.Duration.